registrations/tasks.py

```diff
@@ -375,6 +382,118 @@ def run(self, registration_id, **kwargs):
 validate_subscribe = ValidateSubscribe()
 
 
+class PushRegistrationToJembi(Task):
+    """ Task to push registration data to Jembi
+    """
+    name = "ndoh_hub.registrations.tasks.push_registration_to_jembi"
+    l = get_task_logger(__name__)
+
+    def get_patient_id(self, id_type, id_no=None,
+                       passport_origin=None, mom_msisdn=None):
+        if id_type == 'sa_id':
+            return id_no + "^^^ZAF^NI"
+        elif id_type == 'passport':
+            return id_no + '^^^' + passport_origin.upper() + '^PPN'
+        else:
+            return mom_msisdn.replace('+', '') + '^^^ZAF^TEL'
+
+    def get_subscription_type(self, authority):
+        authority_map = {
+            'unknown': 0,   # NOTE: this was added after the migration
+                            #       to deal with any kind of oddities in the
+                            #       data and how authority & sources are mapped
+            'personal': 1,
+            'chw': 2,
+            'clinic': 3
+        }
+        return authority_map[authority]
+
+    def get_today(self):
+        return datetime.today()
+
+    def get_timestamp(self):
+        return self.get_today().strftime("%Y%m%d%H%M%S")
+
+    def get_dob(self, mom_dob):
+        if mom_dob is not None:
+            return mom_dob.strftime("%Y%m%d")
+        else:
+            return None
+
+    def get_authority_from_source(self, source):
+        """
+        NOTE:   this is a convenience method to map the new "source"
+                back to ndoh-control's "authority" fields to maintain
+                backwards compatibility with existing APIs
+        """
+        return {
+            'CLINIC USSD app': 'clinic',
+            'PUBLIC USSD app': 'personal',
+            'CHW USSD app': 'chw',
+            'NURSE USSD App': 'nurseconnect',
+        }.get(source.name, 'unknown')  # See note at get_subscription_type
```

Comment:
@jpaidoussi I'm unsure about this. The `Source` model has an `authority` attribute, should I be using that? This mapping only applies to old "migrated" records, not for new registrations I think.